### ISSUE TITLE
We can´t Log in (anymore)

### ISSUE THREAD

**Issue by ferkawi** [RocketChat/Rocket.Chat#18128]
### Description:

Our Log in screen dissapeared. We can´t log in and we can´t get some infos(if we were already loged). Hope you can hel me guys, thank you!

### Steps to reproduce:

1. Go to our chat page https://chat.usgonebeyond.com

### Expected behavior:

Log in Screen appear and log in

### Actual behavior:

![image](https://user-images.githubusercontent.com/44171076/86374666-1eb3e600-bc85-11ea-8526-d5ccbacc82a4.png)

### Server Setup Information:

- Version of Rocket.Chat Server: 3.4.0
- Operating System: Ubuntu 16.04
- Deployment Method: docker compose
- Number of Running Instances: 1
- DB Replicaset Oplog: yes
- NodeJS Version: 12.16.1
- MongoDB Version: 4.0.18

### Client Setup Information

- Desktop App or Browser Version: Google Chrome Version 83.0.4103.116 
- Operating System: Windows 10 / MacOS Sierra

### Relevant logs:

![image](https://user-images.githubusercontent.com/44171076/86375061-85d19a80-bc85-11ea-8495-27f5613dfa04.png)
![image](https://user-images.githubusercontent.com/44171076/86375101-9124c600-bc85-11ea-83c3-ad080ac91a10.png)

**Comment by aleadco** [comment]
The same thing happens to us on our server after updating to version 3.4. Please fix that soon. Greetings boys.

**Comment by serviceman** [comment]
+1 having the exact shame issue upgrading from version 3.0.12

**Comment by gannebamm** [comment]
We, too, have this issue.

**Comment by rodrigok** [comment]
@ferkawi can you check the network calls and post some of the answers here so we could check the "Invalid JSON"?

**Comment by rodrigok** [comment]
@gannebamm did you upgrade your installation? Can you inform the previous version and the new one?

**Comment by aleadco** [comment]
> @gannebamm did you upgrade your installation? Can you inform the previous version and the new one?

In my server from 3.3.3 to 3.4 to 3.4.1

**Comment by rodrigok** [comment]
@aleadco can you try to create a collection in your database named `rocketchat_livechat_office_hour` with no records and check if the startup works? My guess it's trying to delete this collection and it doesn't exists

**Comment by aleadco** [comment]
> @ferkawi can you check the network calls and post some of the answers here so we could check the "Invalid JSON"?

a call returns this record, see attached
[log.txt](https://github.com/RocketChat/Rocket.Chat/files/4880304/log.txt)

**Comment by Dizzle397** [comment]
I am also have the same issue. Upgraded from 2.4.11 to 3.4.1, server starts fine. Login form does not appear.

**Comment by rodrigok** [comment]
@aleadco I need your browser's log, the console ones and, if they are the same "Invalid JSON", please grab the HTTP calls' answers as well so we can check what is wrong for you.

**Comment by ggazzo** [comment]
looks good here 🤔 
![image](https://user-images.githubusercontent.com/5263975/86623017-36e27880-bf97-11ea-9caf-ca906a5c533f.png)
![image](https://user-images.githubusercontent.com/5263975/86623099-5d081880-bf97-11ea-8666-ccef587a17cc.png)
![image](https://user-images.githubusercontent.com/5263975/86623171-7b6e1400-bf97-11ea-8ce9-399e363d4367.png)

> ### Description:
> Our Log in screen dissapeared. We can´t log in and we can´t get some infos(if we were already loged). Hope you can hel me guys, thank you!
> 
> ### Steps to reproduce:
> 1. Go to our chat page https://chat.usgonebeyond.com
> 
> ### Expected behavior:
> Log in Screen appear and log in
> 
> ### Actual behavior:
> ![image](https://user-images.githubusercontent.com/44171076/86374666-1eb3e600-bc85-11ea-8526-d5ccbacc82a4.png)

**Comment by aleadco** [comment]
> looks good here 🤔
> ![image](https://user-images.githubusercontent.com/5263975/86623017-36e27880-bf97-11ea-9caf-ca906a5c533f.png)
> ![image](https://user-images.githubusercontent.com/5263975/86623099-5d081880-bf97-11ea-8666-ccef587a17cc.png)
> ![image](https://user-images.githubusercontent.com/5263975/86623171-7b6e1400-bf97-11ea-8ce9-399e363d4367.png)
> 
> > ### Description:
> > Our Log in screen dissapeared. We can´t log in and we can´t get some infos(if we were already loged). Hope you can hel me guys, thank you!
> > ### Steps to reproduce:
> > 
> > 1. Go to our chat page https://chat.usgonebeyond.com
> > 
> > ### Expected behavior:
> > Log in Screen appear and log in
> > ### Actual behavior:
> > ![image](https://user-images.githubusercontent.com/44171076/86374666-1eb3e600-bc85-11ea-8526-d5ccbacc82a4.png)

After updating to 3.4, several instances obtained login errors, not only does it affect the login, it turns out that the changes I make in web administration with the sessions that I have not yet closed are not saved either.

![login](https://user-images.githubusercontent.com/35962168/86623598-dac82600-bf8f-11ea-9dff-6738e64a044e.jpg)

**Comment by aleadco** [comment]
> @aleadco I need your browser's log, the console ones and, if they are the same "Invalid JSON", please grab the HTTP calls' answers as well so we can check what is wrong for you.

I'm on it

**Comment by aleadco** [comment]
@rodrigok This is all I get in a request on my instance.

#### Console in browser Chrome
002b3fae15d3958aa1d6574fb40fc3507e33d688.js?meteor_js_resource=true:formatted:409 Discarding message with invalid JSON undefined
002b3fae15d3958aa1d6574fb40fc3507e33d688.js?meteor_js_resource=true:formatted:409 discarding invalid livedata message null
002b3fae15d3958aa1d6574fb40fc3507e33d688.js?meteor_js_resource=true:formatted:409 Discarding message with invalid JSON undefined
002b3fae15d3958aa1d6574fb40fc3507e33d688.js?meteor_js_resource=true:formatted:409 discarding invalid livedata message null

[rocketchat.error.log](https://github.com/RocketChat/Rocket.Chat/files/4880598/rocketchat.error.log)
[rocketchat.access.log](https://github.com/RocketChat/Rocket.Chat/files/4880599/rocketchat.access.log)
[syslog.log](https://github.com/RocketChat/Rocket.Chat/files/4880603/syslog.log)

**Comment by akurach** [comment]
as for me - problem with using nginx as proxy.
switched to traefik - works well, just try direct access to container http://ip:3000

**Comment by rodrigok** [comment]
@gannebamm your problem should be fixed by this PR https://github.com/RocketChat/Rocket.Chat/pull/18185, lets figure out this other issue related to the login and then we will release version 3.4.2 with those fixes.

**Comment by rodrigok** [comment]
@aleadco we need the requests from your browser to check what it receives as answers that are generating these errors.

Can you open your Developer Tools, reload your page, go to the network tab, right click on any request and click to save as HAR? (Chrome instructions)

![image](https://user-images.githubusercontent.com/234261/86626596-45339300-bf9d-11ea-8bfc-098553b8911b.png)

**Comment by aleadco** [comment]
@rodrigok 

[chat.cneuro.cu.zip](https://github.com/RocketChat/Rocket.Chat/files/4880646/chat.cneuro.cu.zip)

**Comment by aleadco** [comment]
> @rodrigok
> 
[har.zip](https://github.com/RocketChat/Rocket.Chat/files/4880651/har.zip)

**Comment by rodrigok** [comment]
Can you try to change the setting `API_Use_REST_For_DDP_Calls` to false and check if solves your problem?

You can do it via database:
```
db.getCollection('rocketchat_settings').update({_id: 'API_Use_REST_For_DDP_Calls'}, {$set: {value: false}})
```

**Comment by aleadco** [comment]
@rodrigok Problem solved, that was enough. Thank you very much

**Comment by Dizzle397** [comment]
> Can you try to change the setting `API_Use_REST_For_DDP_Calls` to false and check if solves your problem?
> 
> You can do it via database:
> 
> ```
> db.getCollection('rocketchat_settings').update({_id: 'API_Use_REST_For_DDP_Calls'}, {$set: {value: false}})
> ```

This working for me also. Had the exact same issue as @aleadco . Thank you @rodrigok

**Comment by rodrigok** [comment]
Are some of you able to have a 1:1 chat with me at https://open.rocket.chat/direct/rodrigo.nascimento to try to find the root cause and fix it for 3.4.2?

**Comment by serviceman** [comment]
> db.getCollection('rocketchat_settings').update({_id: 'API_Use_REST_For_DDP_Calls'}, {$set: {value: false}})

That worked for me, my chat actually loads now, but now when I try to access the admin panel it quickly closes. On occasion I will get a console error that looks like this. Not sure if it's related to this or not, so I'll post it here. If you want it to have it's own thread I'll make a new one.
![Capture](https://user-images.githubusercontent.com/40190297/86701913-72556500-bfe0-11ea-8f79-a027dc43defd.PNG)

**Comment by alleait** [comment]
I can confirm we are facing the issue above , we stay at 3.3.0 for now until a fix is released.

**Comment by rodrigok** [comment]
@serviceman can you open a new issue for that one?

**Comment by rodrigok** [comment]
@alleait which one? Are you running a staging server where we can debug the issue? Unfortunately we weren't able to reproduce it on our side yet.

**Comment by alleait** [comment]
> @alleait which one? Are you running a staging server where we can debug the issue? Unfortunately we weren't able to reproduce it on our side yet.

We have the blank login screen issue after upgrade.

**Comment by rodrigok** [comment]
@alleait do you have a staging or development server where we could reproduce the problem to investigate? Are you using Nginx as your reverse proxy? If yes can you provide your config?

**Comment by serviceman** [comment]
I have solved this issue in my install without using this flag.
db.getCollection('rocketchat_settings').update({_id: 'API_Use_REST_For_DDP_Calls'}, {$set: {value: false}})

This is a mod_proxy issue. You must have this set in your mod_proxy config.

AllowEncodedSlashes On

As well as changing [P,L] to [P,NE,L] so mod_proxy won't double encode the URLs. This should probably be added to the install and upgrade guides in the documentation.

**Comment by serviceman** [comment]
It cuts off the closing location tag for some reason, so make sure it's added.

**Comment by rodrigok** [comment]
Closed via https://github.com/RocketChat/Rocket.Chat/pull/18222

**Comment by ferkawi** [comment]
I´m sorry that didnt answer before. First solved it downgrading to 3.2.0 but I had today the same Problem (I updated to 3.4.2) and I got the Solution. In our Config was Account->Iframe enabled. Just disabled it and Login Page came back!

Thank you guys for the hel anyways!

Cheers, Nando
